File: lib/models/record.dart
class PatientRecord {
  final String id;
  final String patientId;
  final String doctorId;
  final String doctorName;
  final String date;
  final String diagnosis;
  final String prescription;
  final List<String> attachments;

  PatientRecord({
    required this.id,
    required this.patientId,
    required this.doctorId,
    required this.doctorName,
    required this.date,
    required this.diagnosis,
    required this.prescription,
    this.attachments = const [],
  });

  factory PatientRecord.fromMap(Map<String, dynamic> map) {
    return PatientRecord(
      id: map['_id'] ?? map['id'],
      patientId: map['patientId'] ?? '',
      doctorId: map['doctorId'] ?? '',
      doctorName: map['doctorName'] ?? '',
      date: map['date'] ?? '',
      diagnosis: map['diagnosis'] ?? '',
      prescription: map['prescription'] ?? '',
      attachments: List<String>.from(map['attachments'] ?? []),
    );
  }
}

File: backend/models/Record.js
const mongoose = require('mongoose');

const recordSchema = new mongoose.Schema({
    patientId: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User',
        required: true
    },
    doctorId: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User',
        required: true
    },
    doctorName: {
        type: String,
        required: true
    },
    date: {
        type: String, // format YYYY-MM-DD
        required: true
    },
    diagnosis: {
        type: String,
        required: true
    },
    prescription: {
        type: String,
        default: ''
    },
    attachments: [{
        type: String
    }]
}, {
    timestamps: true
});

module.exports = mongoose.model('Record', recordSchema);
